testing/scenario_app/lib/main.dart

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+import 'dart:ui';
+
+import 'src/animated_color_square.dart';
+import 'src/scenario.dart';
+
+Map<String, Scenario> _scenarios = <String, Scenario>{
+  'animated_color_square': AnimatedColorSquareScenario(window),
+};
+
+Scenario _currentScenario = _scenarios['animated_color_square'];
+
+void main() {
+  window
+    ..onPlatformMessage = _handlePlatformMessage
```

Comment:
It looks like we're emulating some standard Flutter behavior here. It would probably be a good idea to add comments explaining what these behaviors are, and given that we're carefully reproducing only a few behaviors, it would be helpful to know why each of these are provided and others are not.

Reply:
I'm not implementing any real Flutter framework behavior here.  I'm setting up a handler for platform messages as a stub for when we fill these out with more behavior/tests.  I'm also creating a way to write multiple Dart based scenarios for testing that just handles register and forwarding some critical things from the Window.

Reply:
But aren't you implementing these things because they are part of a standard Flutter contract that the embedding expects Flutter to fulfill?

Reply:
The embedding should only care about dart:ui, not the full Flutter framework.  I'm not sure which you mean in this question.

I'm setting up enough so I can draw things and notify the embedding that I'm ready.  It's missing a lot that the framework does.  It's also sending a message the framework doesn't.

Reply:
Ok, the framework aside, I guess what I'm getting at is this. Let's imagine a developer comes into this code to see what's going on so they can add a test. They make their way into `main()` and they see this configuration logic. That developer may lack a lot of the contextual knowledge that all of us on this thread have right now. They may have a murky understanding of the exact boundary between the embedding and the framework, they may not be particularly familiar with Window, or even with the special area that is dart:ui. 

So I'm wondering if it would be worth some expository here to provide at least an actionable level of familiarity with the hooks that are being implemented on `window`.  We could have implemented a lot more behavior here, or we could have implemented less/no behavior here, but this seems to be a goldilocks position, and it might be worth explaining why that is. But that's just a suggestion.